.github/workflows/release.yml

```diff
@@ -0,0 +1,315 @@
+        uses: jamessingleton/is-organization-member@1.0.1
+        with:
+          organization: viamrobotics
+          username: ${{ github.actor }}
+          token:  ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Cancelling - user not part of organization
+        uses: andymckay/cancel-action@0.2
+        if: |
+          steps.is_organization_member.outputs.result == 'false'
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Which Version
+        id: which_version
+        shell: bash
+        run: echo "version=$(grep 'set(CMAKE_PROJECT_VERSION' CMakeLists.txt | sed 's/)//' | awk '{print $2}')" >> $GITHUB_OUTPUT
+
+      - name: Check if release exists
+        uses: cardinalby/git-get-release-action@1.2.4
+        id: release_exists
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          releaseName: v${{ steps.which_version.outputs.version }}
+          doNotFailIfNotFound: 'true'
+
+      - name: Cancelling - release already exists
+        uses: andymckay/cancel-action@0.2
+        if: steps.release_exists.outputs.id != ''
+
+      - name: Commit + Push
+        id: commit
+        uses: EndBug/add-and-commit@v9.0.0
+        with:
+          default_author: github_actions
+          message: Bumping version to v${{ steps.which_version.outputs.version }} [skip ci]
+
+  build_macos:
+    if: github.repository_owner == 'viamrobotics'
+    needs: [prepare]
+    runs-on: [self-hosted, ARM64, macOS]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: aarch64-apple-darwin
+            platform: macosx_arm64
+          - target: x86_64-apple-darwin
+            platform: macosx_x86_64
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.prepare.outputs.sha }}
+
+      - name: Install dependencies
+        run: |
+          brew install abseil cmake boost grpc protobuf xtensor
+          brew install pkg-config ninja buf
+
+      - name: Setup build directory
+        run: mkdir builds
+
+      - name: Build
+        run: |
+          export BUF_CONFIG_DIR=$(mktemp -d)
+          mkdir build
+          pushd build/
+          cmake .. -G Ninja
+          ninja all && ninja install
+          popd
+
+      - name: Copy
+        run: |
+          cp build/install/lib/libviam_rust_utils.a builds/libviam_rust_utils-${{ matrix.platform }}.a
+          cp build/install/lib/libviamsdk.${{ needs.prepare.outputs.version }}.dylib builds/libviamsdk-${{ matrix.platform }}.dylib
+          cp build/install/lib/libviamapi.${{ needs.prepare.outputs.version }}.dylib builds/libviamapi-${{ matrix.platform }}.dylib
+      - name: Correct install path
+        run: |
+          install_name_tool -id "@rpath/libviamsdk.dylib" builds/libviamsdk-${{ matrix.platform }}.dylib
+          install_name_tool -id "@rpath/libviamapi.dylib" builds/libviamapi-${{ matrix.platform }}.dylib
```

Comment:
I think you don't need to do this. You should just be able to use `cmake` in `install` mode and it'll do it for you:

```
cmake --install build --prefix builds/libviamsdk-${{ matrix.platform }}
```
It optionally takes a component, so if you really do want to restrict to just the runtime libs and not the dev components like headers and cmake support files, you could do:
```
cmake --install build --prefix builds/libviamsdk-${{ matrix.platform }} --component=viam-cpp-sdk_runtime
```